src/peergos/shared/user/ProfilePaths.java

```diff
@@ -166,6 +166,20 @@
         return serializeAndSet(WEBROOT, webroot, String::getBytes, user);
     }
 
+    @JsMethod
+    public static CompletableFuture<Boolean> unPublish(UserContext user) {
+        return getWebRoot(user.username, user)
```

Comment:
before rotating the keys we should remove the published cap to the dir. 
This can go on UserContext:
public CompletableFuture<CommittedWriterData> removePublicCap(String path) {
        return writeSynchronizer.applyUpdate(signer.publicKeyHash, signer, (wd, tid) -> {
            Optional<Multihash> publicData = wd.publicData;
            if (publicData.isEmpty())
                return Futures.of(wd);
            return network.dhtClient.get(publicData.get())
                    .thenCompose(rootCbor -> InodeFileSystem.build(rootCbor.get(), crypto.hasher, network.dhtClient))
                    .thenCompose(pubCaps -> pubCaps.removeCap(signer.publicKeyHash, signer, path, tid))
                    .thenCompose(updated -> network.dhtClient.put(signer.publicKeyHash, signer, updated.serialize(), crypto.hasher, tid))
                    .thenApply(newRoot -> wd.withPublicRoot(newRoot));
        }).thenApply(v -> v.get(signer));
    }